table/block.h

```diff
@@ -206,42 +206,14 @@ class Block {
 
 class BlockIter : public InternalIterator {
  public:
-  // Object created using this constructor will behave like an iterator
-  // against an empty block. The state after the creation: Valid()=false
-  // and status() is OK.
-  BlockIter()
```

Comment:
Can you define the default constructor as protected so no one could mistakenly create object off BlockIter?

Reply:
Sure I can delete it. But using it will have a compile error anyway because there is undefined virtual functions.

Reply:
oh cool. that is sufficient then.